### PR TITLE
Expose iteration over verification relationship fields

### PR DESCRIPTION
**Pull request by abdulmth**
# Description of change
Add an optional scope parameter to the method getter on DID Document implementations to allow iterating over particular verification relationship fields.

## Links to any relevant issues
`fixes issue #989`

## Type of change
Add an `x` to the boxes that are relevant to your changes.

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] Enhancement (a non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Fix

### CODE REVIEW

**Review by PhilippGackstatter — APPROVED** [review]
Looks good to me!

I tried a different approach with `Box<dyn Iterator>` but it produces a 6 KiB larger Wasm binary, so the `Vec` approach seems preferable despite more allocations and going through the entire iterator upfront.

For reference:

```rust
pub fn methods<'this>(
    &'this self,
    scope: Option<MethodScope>,
  ) -> Box<dyn Iterator<Item = &VerificationMethod<D, U>> + 'this>
  where
    D: DID,
  {
    if let Some(scope) = scope {
      match scope {
        MethodScope::VerificationMethod => Box::new(self.verification_method().iter()),
        MethodScope::VerificationRelationship(MethodRelationship::AssertionMethod) => Box::new(
          self
            .assertion_method()
            .iter()
            .filter_map(|mr| self.resolve_method_ref(mr)),
        ),
        MethodScope::VerificationRelationship(MethodRelationship::Authentication) => Box::new(
          self
            .authentication()
            .iter()
            .filter_map(|mr| self.resolve_method_ref(mr)),
        ),
        MethodScope::VerificationRelationship(MethodRelationship::CapabilityDelegation) => Box::new(
          self
            .capability_delegation()
            .iter()
            .filter_map(|mr| self.resolve_method_ref(mr)),
        ),
        MethodScope::VerificationRelationship(MethodRelationship::CapabilityInvocation) => Box::new(
          self
            .capability_invocation()
            .iter()
            .filter_map(|mr| self.resolve_method_ref(mr)),
        ),
        MethodScope::VerificationRelationship(MethodRelationship::KeyAgreement) => {
          Box::new(self.key_agreement().iter().filter_map(|mr| self.resolve_method_ref(mr)))
        }
      }
    } else {
      Box::new(self.all_methods())
    }
  }
```

File: identity_credential/src/validator/credential_validator.rs [hunk 1]
```diff
@@ -521,7 +521,7 @@ mod tests {
     issuer_doc
       .signer(issuer_key.private())
       .options(ProofOptions::default())
-      .method(issuer_doc.methods().next().unwrap().id())
+      .method(issuer_doc.methods(None).get(0).unwrap().id())
```
Comment (cycraig) [inline identity_credential/src/validator/credential_validator.rs:524]: Nit: prefer `.first()` over `.get(0)` in general. Doesn't have to be changed.

**Review by cycraig — APPROVED** [review]
Looks good to me.

File: identity_iota_core/src/document/iota_document.rs [hunk 2]
```diff
@@ -161,9 +161,11 @@ impl IotaDocument {
   // Verification Methods
   // ===========================================================================
 
-  /// Returns an iterator over all [`IotaVerificationMethod`] in the DID Document.
-  pub fn methods(&self) -> impl Iterator<Item = &IotaVerificationMethod> {
-    self.document.methods()
+  /// Returns a `Vec` of verification methods references whose verification relationship matches `scope`.
```
Comment (cycraig) [inline identity_iota_core/src/document/iota_document.rs:164]:
```suggestion
  /// Returns a `Vec` of verification methods reference whose verification relationship matches `scope`.
```

**Pull request merged**